src/main/scala/com/cloudera/sparkts/TimeSeries.scala

```diff
@@ -169,6 +169,55 @@ class TimeSeries[K](val index: DateTimeIndex, val data: DenseMatrix,
   }
 
   /**
+    * Returns a TimeSeries where each individual time series is differenced with the given time
+    * frequency.
+    *
+    * The new TimeSeries will be missing data (truncated) up to the first instance of the provided
+    * lag period. This operation otherwise preserves the date/time index: no resampling occurs.
+    *
+    * It handles the NaN values in the following way:
+    *
+    * If we are differencing [x(t) - x(t-k)], for example, where k is the lag period, then
+    * if x(t) is NaN, the result is NaN. If x(t-k) is NaN but x(t) is not, however, it looks for
+    * the most recent non-NaN value before t-k and uses that to calculate the difference.
+    */
+  def timeDifferences(baseFrequency: Frequency): TimeSeries[K] = {
```

Comment:
I think it probably makes sense just to call this `differences`, or `differencesByFrequency`.  The `time` suffix doesn't really accurately contrast it with the other `differences` method.